syntax = "proto3";
package bms;


enum bms_board_status_cellboard0_balancing_status {
	bms_board_status_cellboard0_balancing_status_OFF = 0;
	bms_board_status_cellboard0_balancing_status_DISCHARGE = 1;
}
enum bms_board_status_cellboard1_balancing_status {
	bms_board_status_cellboard1_balancing_status_OFF = 0;
	bms_board_status_cellboard1_balancing_status_DISCHARGE = 1;
}
enum bms_board_status_cellboard2_balancing_status {
	bms_board_status_cellboard2_balancing_status_OFF = 0;
	bms_board_status_cellboard2_balancing_status_DISCHARGE = 1;
}
enum bms_board_status_cellboard3_balancing_status {
	bms_board_status_cellboard3_balancing_status_OFF = 0;
	bms_board_status_cellboard3_balancing_status_DISCHARGE = 1;
}
enum bms_board_status_cellboard4_balancing_status {
	bms_board_status_cellboard4_balancing_status_OFF = 0;
	bms_board_status_cellboard4_balancing_status_DISCHARGE = 1;
}
enum bms_board_status_cellboard5_balancing_status {
	bms_board_status_cellboard5_balancing_status_OFF = 0;
	bms_board_status_cellboard5_balancing_status_DISCHARGE = 1;
}


message BOARD_STATUS_CELLBOARD0{
	bool errors_CAN_COMM = 1;
	bool errors_LTC_COMM = 2;
	bool errors_TEMP_COMM_0 = 3;
	bool errors_TEMP_COMM_1 = 4;
	bool errors_TEMP_COMM_2 = 5;
	bool errors_TEMP_COMM_3 = 6;
	bool errors_TEMP_COMM_4 = 7;
	bool errors_TEMP_COMM_5 = 8;
	bool errors_OPEN_WIRE = 9;
	bms_board_status_cellboard0_balancing_status balancing_status = 10;
	uint64 _inner_timestamp = 11;
}
message BOARD_STATUS_CELLBOARD1{
	bool errors_CAN_COMM = 1;
	bool errors_LTC_COMM = 2;
	bool errors_TEMP_COMM_0 = 3;
	bool errors_TEMP_COMM_1 = 4;
	bool errors_TEMP_COMM_2 = 5;
	bool errors_TEMP_COMM_3 = 6;
	bool errors_TEMP_COMM_4 = 7;
	bool errors_TEMP_COMM_5 = 8;
	bool errors_OPEN_WIRE = 9;
	bms_board_status_cellboard1_balancing_status balancing_status = 10;
	uint64 _inner_timestamp = 11;
}
message BOARD_STATUS_CELLBOARD2{
	bool errors_CAN_COMM = 1;
	bool errors_LTC_COMM = 2;
	bool errors_TEMP_COMM_0 = 3;
	bool errors_TEMP_COMM_1 = 4;
	bool errors_TEMP_COMM_2 = 5;
	bool errors_TEMP_COMM_3 = 6;
	bool errors_TEMP_COMM_4 = 7;
	bool errors_TEMP_COMM_5 = 8;
	bool errors_OPEN_WIRE = 9;
	bms_board_status_cellboard2_balancing_status balancing_status = 10;
	uint64 _inner_timestamp = 11;
}
message BOARD_STATUS_CELLBOARD3{
	bool errors_CAN_COMM = 1;
	bool errors_LTC_COMM = 2;
	bool errors_TEMP_COMM_0 = 3;
	bool errors_TEMP_COMM_1 = 4;
	bool errors_TEMP_COMM_2 = 5;
	bool errors_TEMP_COMM_3 = 6;
	bool errors_TEMP_COMM_4 = 7;
	bool errors_TEMP_COMM_5 = 8;
	bool errors_OPEN_WIRE = 9;
	bms_board_status_cellboard3_balancing_status balancing_status = 10;
	uint64 _inner_timestamp = 11;
}
message BOARD_STATUS_CELLBOARD4{
	bool errors_CAN_COMM = 1;
	bool errors_LTC_COMM = 2;
	bool errors_TEMP_COMM_0 = 3;
	bool errors_TEMP_COMM_1 = 4;
	bool errors_TEMP_COMM_2 = 5;
	bool errors_TEMP_COMM_3 = 6;
	bool errors_TEMP_COMM_4 = 7;
	bool errors_TEMP_COMM_5 = 8;
	bool errors_OPEN_WIRE = 9;
	bms_board_status_cellboard4_balancing_status balancing_status = 10;
	uint64 _inner_timestamp = 11;
}
message BOARD_STATUS_CELLBOARD5{
	bool errors_CAN_COMM = 1;
	bool errors_LTC_COMM = 2;
	bool errors_TEMP_COMM_0 = 3;
	bool errors_TEMP_COMM_1 = 4;
	bool errors_TEMP_COMM_2 = 5;
	bool errors_TEMP_COMM_3 = 6;
	bool errors_TEMP_COMM_4 = 7;
	bool errors_TEMP_COMM_5 = 8;
	bool errors_OPEN_WIRE = 9;
	bms_board_status_cellboard5_balancing_status balancing_status = 10;
	uint64 _inner_timestamp = 11;
}
message TEMPERATURES_CELLBOARD0{
	uint32 start_index = 1;
	double temp0 = 2;
	double temp1 = 3;
	double temp2 = 4;
	double temp3 = 5;
	double temp4 = 6;
	double temp5 = 7;
	uint64 _inner_timestamp = 8;
}
message TEMPERATURES_CELLBOARD1{
	uint32 start_index = 1;
	double temp0 = 2;
	double temp1 = 3;
	double temp2 = 4;
	double temp3 = 5;
	double temp4 = 6;
	double temp5 = 7;
	uint64 _inner_timestamp = 8;
}
message TEMPERATURES_CELLBOARD2{
	uint32 start_index = 1;
	double temp0 = 2;
	double temp1 = 3;
	double temp2 = 4;
	double temp3 = 5;
	double temp4 = 6;
	double temp5 = 7;
	uint64 _inner_timestamp = 8;
}
message TEMPERATURES_CELLBOARD3{
	uint32 start_index = 1;
	double temp0 = 2;
	double temp1 = 3;
	double temp2 = 4;
	double temp3 = 5;
	double temp4 = 6;
	double temp5 = 7;
	uint64 _inner_timestamp = 8;
}
message TEMPERATURES_CELLBOARD4{
	uint32 start_index = 1;
	double temp0 = 2;
	double temp1 = 3;
	double temp2 = 4;
	double temp3 = 5;
	double temp4 = 6;
	double temp5 = 7;
	uint64 _inner_timestamp = 8;
}
message TEMPERATURES_CELLBOARD5{
	uint32 start_index = 1;
	double temp0 = 2;
	double temp1 = 3;
	double temp2 = 4;
	double temp3 = 5;
	double temp4 = 6;
	double temp5 = 7;
	uint64 _inner_timestamp = 8;
}
message VOLTAGES_CELLBOARD0{
	uint32 start_index = 1;
	double voltage0 = 2;
	double voltage1 = 3;
	double voltage2 = 4;
	uint64 _inner_timestamp = 5;
}
message VOLTAGES_CELLBOARD1{
	uint32 start_index = 1;
	double voltage0 = 2;
	double voltage1 = 3;
	double voltage2 = 4;
	uint64 _inner_timestamp = 5;
}
message VOLTAGES_CELLBOARD2{
	uint32 start_index = 1;
	double voltage0 = 2;
	double voltage1 = 3;
	double voltage2 = 4;
	uint64 _inner_timestamp = 5;
}
message VOLTAGES_CELLBOARD3{
	uint32 start_index = 1;
	double voltage0 = 2;
	double voltage1 = 3;
	double voltage2 = 4;
	uint64 _inner_timestamp = 5;
}
message VOLTAGES_CELLBOARD4{
	uint32 start_index = 1;
	double voltage0 = 2;
	double voltage1 = 3;
	double voltage2 = 4;
	uint64 _inner_timestamp = 5;
}
message VOLTAGES_CELLBOARD5{
	uint32 start_index = 1;
	double voltage0 = 2;
	double voltage1 = 3;
	double voltage2 = 4;
	uint64 _inner_timestamp = 5;
}
message BALANCING{
	uint32 board_index = 1;
	bool cells_CELL0 = 2;
	bool cells_CELL1 = 3;
	bool cells_CELL2 = 4;
	bool cells_CELL3 = 5;
	bool cells_CELL4 = 6;
	bool cells_CELL5 = 7;
	bool cells_CELL6 = 8;
	bool cells_CELL7 = 9;
	bool cells_CELL8 = 10;
	bool cells_CELL9 = 11;
	bool cells_CELL10 = 12;
	bool cells_CELL11 = 13;
	bool cells_CELL12 = 14;
	bool cells_CELL13 = 15;
	bool cells_CELL14 = 16;
	bool cells_CELL15 = 17;
	bool cells_CELL16 = 18;
	bool cells_CELL17 = 19;
	uint64 _inner_timestamp = 20;
}
message FW_UPDATE{
	uint32 board_index = 1;
	uint64 _inner_timestamp = 2;
}
message FLASH_CELLBOARD_0_TX{
	uint64 _inner_timestamp = 1;
}
message FLASH_CELLBOARD_0_RX{
	uint64 _inner_timestamp = 1;
}
message FLASH_CELLBOARD_1_TX{
	uint64 _inner_timestamp = 1;
}
message FLASH_CELLBOARD_1_RX{
	uint64 _inner_timestamp = 1;
}
message FLASH_CELLBOARD_2_TX{
	uint64 _inner_timestamp = 1;
}
message FLASH_CELLBOARD_2_RX{
	uint64 _inner_timestamp = 1;
}
message FLASH_CELLBOARD_3_TX{
	uint64 _inner_timestamp = 1;
}
message FLASH_CELLBOARD_3_RX{
	uint64 _inner_timestamp = 1;
}
message FLASH_CELLBOARD_4_TX{
	uint64 _inner_timestamp = 1;
}
message FLASH_CELLBOARD_4_RX{
	uint64 _inner_timestamp = 1;
}
message FLASH_CELLBOARD_5_TX{
	uint64 _inner_timestamp = 1;
}
message FLASH_CELLBOARD_5_RX{
	uint64 _inner_timestamp = 1;
}


message Pack{
	repeated BOARD_STATUS_CELLBOARD0 BOARD_STATUS_CELLBOARD0 = 1;
	repeated BOARD_STATUS_CELLBOARD1 BOARD_STATUS_CELLBOARD1 = 2;
	repeated BOARD_STATUS_CELLBOARD2 BOARD_STATUS_CELLBOARD2 = 3;
	repeated BOARD_STATUS_CELLBOARD3 BOARD_STATUS_CELLBOARD3 = 4;
	repeated BOARD_STATUS_CELLBOARD4 BOARD_STATUS_CELLBOARD4 = 5;
	repeated BOARD_STATUS_CELLBOARD5 BOARD_STATUS_CELLBOARD5 = 6;
	repeated TEMPERATURES_CELLBOARD0 TEMPERATURES_CELLBOARD0 = 7;
	repeated TEMPERATURES_CELLBOARD1 TEMPERATURES_CELLBOARD1 = 8;
	repeated TEMPERATURES_CELLBOARD2 TEMPERATURES_CELLBOARD2 = 9;
	repeated TEMPERATURES_CELLBOARD3 TEMPERATURES_CELLBOARD3 = 10;
	repeated TEMPERATURES_CELLBOARD4 TEMPERATURES_CELLBOARD4 = 11;
	repeated TEMPERATURES_CELLBOARD5 TEMPERATURES_CELLBOARD5 = 12;
	repeated VOLTAGES_CELLBOARD0 VOLTAGES_CELLBOARD0 = 13;
	repeated VOLTAGES_CELLBOARD1 VOLTAGES_CELLBOARD1 = 14;
	repeated VOLTAGES_CELLBOARD2 VOLTAGES_CELLBOARD2 = 15;
	repeated VOLTAGES_CELLBOARD3 VOLTAGES_CELLBOARD3 = 16;
	repeated VOLTAGES_CELLBOARD4 VOLTAGES_CELLBOARD4 = 17;
	repeated VOLTAGES_CELLBOARD5 VOLTAGES_CELLBOARD5 = 18;
	repeated BALANCING BALANCING = 19;
	repeated FW_UPDATE FW_UPDATE = 20;
	repeated FLASH_CELLBOARD_0_TX FLASH_CELLBOARD_0_TX = 21;
	repeated FLASH_CELLBOARD_0_RX FLASH_CELLBOARD_0_RX = 22;
	repeated FLASH_CELLBOARD_1_TX FLASH_CELLBOARD_1_TX = 23;
	repeated FLASH_CELLBOARD_1_RX FLASH_CELLBOARD_1_RX = 24;
	repeated FLASH_CELLBOARD_2_TX FLASH_CELLBOARD_2_TX = 25;
	repeated FLASH_CELLBOARD_2_RX FLASH_CELLBOARD_2_RX = 26;
	repeated FLASH_CELLBOARD_3_TX FLASH_CELLBOARD_3_TX = 27;
	repeated FLASH_CELLBOARD_3_RX FLASH_CELLBOARD_3_RX = 28;
	repeated FLASH_CELLBOARD_4_TX FLASH_CELLBOARD_4_TX = 29;
	repeated FLASH_CELLBOARD_4_RX FLASH_CELLBOARD_4_RX = 30;
	repeated FLASH_CELLBOARD_5_TX FLASH_CELLBOARD_5_TX = 31;
	repeated FLASH_CELLBOARD_5_RX FLASH_CELLBOARD_5_RX = 32;
}

